cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/table/field/FieldValidator.java

```diff
@@ -59,6 +60,35 @@ public void validateField(Field<?> field) throws InvalidFieldException {
     }
   }
 
+  /**
+   * Validate if the given range matches the schema. The given range is invalid if:
+   * its field is not present in the given schema,
+   * its field type is different from the given schema,
+   * if its field is a primary key but the given value is null,
+   * or it does not contain a primary key.
+   *
+   * @param range the range to validate
+   * @throws InvalidFieldException if the field does not pass the validation
+   */
+  public void validateRange(Range range) throws InvalidFieldException {
```

Comment:
nit: validateScanRange or validateQueryRange. Non-pk ranges can be useful for other usecases